up-l1/someip.adoc

```diff
@@ -0,0 +1,221 @@
+|`DATA_LOSS` |`E_MALFORMED_MESSAGE`
+|`INTERNAL` |`E_NOT_REACHABLE`
+|===
+
+
+=== UPayload
+
+`UPayload` hosts the application layer data that is being sent between devices. `UPayload` structure, defined in link:../basics/upayload.adoc[UPayload] contains either a copy or reference to the actual payload that is to be sent. `UPayload` also contains `UPayloadFormat` used to give a hint of the payload format (protobuf serialized, SOME/IP format, TEXT, RAW, etc...). The SOME/IP specifications however does not have an equivalent field for `UPayloadFormat` as it is assumed that the payload is serialized in the format that the other end knows how to deserialize (i.e. it is fixed per topic). As such, when converting between uProtocol and SOME/IP, the `UPayloadFormat` field *SHOULD* be ignored (left at the default of `UMESSAGE_TYPE_UNSPECIFIED`).
+
+
+=== uTransport
+
+Mapping of uTransport APIs to SOME/IP specific library APIs shall not be covered in this document given there are multiple open source libraries available for SOME/IP.
+
+
+== Application Layer Translation
+
+Application (or message payload) translation is the process of converting SOME/IP-SD subscription and discovery messages, to/from uDiscovery and uSubscription Messages. 
+
+=== uSubscription
+
+The following section will elaborate only on the translation of uSubscription messages to/from SOME/IP-SD messages. Subscription state (persistent or not) is handled in the uSubscription services and not at the transport layer or this component.
+
+The following section we will elaborate on how Eventgroup Entry types (5.1.3.1) are mapped to uSubscription messages for the subscribe
+and unsubscribe flows. 
+
+==== Common Fields
+
+<<common-field-mappings>> table below illustrates the common SOME/IP-SD EventGroup Entry fields that are present in for all SOME/IP-SD Eventgroup entry types (`SubscribeEventgroup`, `SubscribeEventGroupAck`, `SubscribeEventgroupNack`, `StopSubscribeEventGroup`). These fields are then mapped to uProtocol `UUri` attributes usined in uProtocol `UMessage` for performing subscription operations. 
+
+.Common Field Mappings
+[#common-field-mappings, width=70%]
+|===
+|Eventgroup Entry Field |UUri
+
+|Service ID |`UEntity.id`
+
+|Major Version | `UEntity.major_version`
+
+|Eventgroup ID | `UResource.id`
+
+|===
+
+NOTE: `UUri.authority.ip` is translated to/from IPv4 (and/or IPv6) Endpoint Option of the SOME/IP-SD message
+
+==== EventGroup Entry Type Mapping
+
+.EventGroup Entry Type Mapping
+[#eventgroup-entry-mapping, cols="1,1,2"]
+|===
+|Eventgroup Entry Type |uSubscription Message | Additional Details
+
+|`SubscribeEventGroup` 
+|`SubscriptionRequest` 
+a|The message is used to subscribe to a topic.
+
+* Instance ID *SHALL* assume to be 0
+* *SHALL* assume version 0 to match uProtocol backwards compatibility requirements
+* TTL (`SubscribeAttributes.ttl`) *SHALL* set to 0xFFFFFF to indicate that the subscription should remain for the duration of the ignition cycle
```

Comment:
Remark about vsomeip: there is global Service Discovery TTL (static config) and it is used for all subscriptions. It's not possible to have it per service or dynamic. Probably it should match remote service shortest TTL

Reply:
@int0x27 , given your comment, should we assume live forever?